stdlib/public/core/Policy.swift

```diff
@@ -15,8 +15,26 @@
 //===----------------------------------------------------------------------===//
 // Standardized uninhabited type
 //===----------------------------------------------------------------------===//
-/// The return type of functions that do not return normally, that is, a type
-/// with no values.
+/// The return type of functions that cannot return a value.
+///
+/// `Never` is a type with no values and cannot be constructed, allowing you
+/// to represent a state in your program that is impossible to reach during
+/// its execution.
+///
+///     // An infinite loop will never terminate.
+///     func forever() -> Never {
+///         while true {
+///             print("I will print forever and will not return.")
+///         }
+///     }
```

Comment:
Your `forever()` example could move further down, alongside `crashAndBurn()`.

Reply:
Added a snippet about *nonreturning* functions from the Swift Reference, so I've placed it below that. It seems this makes the most sense and separates each example snippet.